File: algorithms/661. Image Smoother/solution.cc
class Solution {
public:
    vector<vector<int>> imageSmoother(vector<vector<int>>& M) {
        int m = M.size(), n = M[0].size();
        vector<vector<int>> r(m, vector<int> (n));
        for ( int i = 0; i < m; ++i ) {
            for ( int j = 0; j < n; ++j ) {
                int cnt = 0;
                for ( int ni = i - 1; ni < i + 2; ++ni ) {
                    for ( int nj = j - 1; nj < j + 2; ++nj ) {
                        if ( 0 <= ni && ni < m && 0 <= nj && nj < n ) {
                            ++cnt;
                            r[i][j] += M[ni][nj];
                        }
                    }
                }
                r[i][j] /= cnt;
            }
        }
        return r;
    }
};

// 202 / 202 test cases passed.
// Status: Accepted
// Runtime: 172 ms

File: algorithms/661. Image Smoother/solution.py
class Solution(object):
    def imageSmoother(self, M):
        """
        :type M: List[List[int]]
        :rtype: List[List[int]]
        """
        m, n = len(M), len(M[0])
        r = [[0] * n for i in xrange(m)]
        for i in xrange(m):
            for j in xrange(n):
                c = 0
                for ni in xrange(i-1, i+2):
                    for nj in xrange(j-1, j+2):
                        if 0 <= ni < m and 0 <= nj < n:
                            r[i][j] += M[ni][nj]
                            c += 1
                r[i][j] /= c
        return r

# 202 / 202 test cases passed.
# Status: Accepted
# Runtime: 939 ms
